runtime/src/comm/ofi/comm-ofi.c

```diff
@@ -4066,6 +4081,39 @@ void amRequestShutdown(c_nodeid_t node) {
   amRequestCommon(node, &req, sizeof(req.b), false, NULL);
 }
 
+// amRequestFAMOResult
+//
+// This function is invoked by a fetching AMO handler to return the result
+// back to the initiating locale. The scenario is that locale A has sent a
+// fetching AMO to locale B via an AM. The AM handler on B invokes this
+// function to send an AM back to A that causes A to store the result of the
+// fetching AMO and set the pAmDone flag.
+
+static inline
+void amRequestFAMOResult(c_nodeid_t node,
+                  chpl_amo_datum_t *result,
+                  enum fi_datatype ofiType, // type of the result
+                  void *targetAddr, // address of result on remote node
+                  size_t size,       // size of the result
+                  amDone_t *pAmDone) // address of pAmDone on remote node
+{
+
+  DBG_PRINTF(DBG_AM | DBG_AM_SEND, "amRequestFAMOResult");
+  struct perTxCtxInfo_t* tcip;
+  CHK_TRUE((tcip = tciAlloc()) != NULL);
+
+  amRequest_t req = { .famo_result = { .b = { .op = am_opFAMOResult,
+                                      .node = chpl_nodeID,
+                                      .pAmDone = pAmDone, },
+                               .size = size,
+                               .ofiType = ofiType,
+                               .obj = targetAddr,}, };
+
+  memcpy(&req.famo_result.object, result, size);
```

Comment:
It probably won't be any slower, and seems like it would be clearer, to just include `.result = *result` in the struct assignment above instead of doing a separate `memcpy()`, since both of those are `chpl_amo_datum_t` values.